### PR TITLE
[tl] Add tagalog deinflections

### PR DESCRIPTION
**Pull request by Casheeew**
Add deinflections (mainly affixes) for Tagalog.

Sources:
https://learningtagalog.com/grammar/nouns/noun_affixes.html
https://learningtagalog.com/grammar/verbs/verb_affixes.html
https://learningtagalog.com/grammar/adjectives/adjective_affixes.html
https://en.wikipedia.org/wiki/Tagalog_grammar

### CODE REVIEW

**Comment by jamesmaa** [comment]
If we don't have tagalog language experts here we can merge this in a couple of days. I can make an open call in the tagalog subreddit in the meantime

**Review by poypoyan — COMMENTED** [review]
I do not completely understand the code, but yep, only one for now.

Aside from the comment/s in the subreddit (about infixes) I like to comment that there are 2 rule versions about _nakaka-_ and _nakakapang-_: the casual and the formal. The Learning Tagalog one is the casual rule. On the other hand, the formal rule is _naka-_/_nakapang_ + rep1 (first syllable repeated). For example,

nakakatulong (casual) = naka*tu*tulong (formal)
nakakapanghina (casual) = nakapang*hi*hina (formal).

Afaik, there is no analog for formal rule for _pagkaka-_ and _magkaka-_: "pagkasasabi" is not a word.

But yea, other than these, LGTM for now 😅.

**Comment by cashewnuttynuts** [comment]
Thanks very much for the feedback! I will add those changes

**Comment by poypoyan** [comment]
What made me comment about the 2 rules is the test for _kaka-_: "kaiisip" is formal, while "kakaisip" is casual. I see that Learning Tagalog included both rules for that.

I'm sorry I forgot to mention this: _pang-_ is also used as ordinal prefix. This is casual, while _ika-_ is formal. And again, it is pangalawa and pangatlo for second and third.

**Review by poypoyan — COMMENTED** [review]
Just corrections...

Thank you very much!

File: ext/js/language/tl/tagalog-transforms.js [hunk 1]
```diff
@@ -0,0 +1,718 @@
...
+            ],
+        },
+        'pang-': {
+            name: 'pang-',
+            rules: [
+                prefixInflection('pang', '', [], ['n', 'adj']),
+                ...[...'dlrst'].map((v) => prefixInflection(`pan${v}`, `${v}`, [], ['n', 'adj'])),
+                ...[...'bp'].map((v) => prefixInflection(`pam${v}`, `${v}`, [], ['n', 'adj'])),
+                wholeWordInflection('pangalawa', 'dawa', [], ['num']),
+                wholeWordInflection('pangatlo', 'dawa', [], ['num']),
```
Comment (poypoyan) [inline ext/js/language/tl/tagalog-transforms.js:184]: 'dalawa'
'tatlo'

File: test/language/tagalog-transforms.test.js [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+            {term: 'kain', source: 'kainin', rule: 'n', reasons: ['-in']},
+            {term: 'ako', source: 'akong', rule: 'n', reasons: ['-ng']},
+        ],
+    },
+    {
+        category: 'irregulars',
+        valid: true,
+        tests: [
+            {term: 'dawa', source: 'ikalawa', rule: 'num', reasons: ['ika-']},
+            {term: 'dawa', source: 'pangalawa', rule: 'num', reasons: ['pang-']},
```
Comment (poypoyan) [inline test/language/tagalog-transforms.test.js:57]: Both should be 'dalawa'.

**Review by poypoyan — COMMENTED** [review]
Just corrections.

Thank you very much for this work!

**Review by StefanVukovic99 — APPROVED** [review]

**Pull request merged**